lstchain/calib/camera/pixel_threshold_estimation.py

```diff
@@ -79,3 +83,78 @@ def get_unusable_pixels(dl1_path, pedestal_id):
+    increases the cleaning picture threshold of a pixel to, say, 2.5 standard
+    deviations (or the number in "sigma" below) above its pedestal mean,
+    hence pixels illuminated by stars get a higher threshold, and so we avoid
+    too many spurious signals from the starlight fluctuations. The downside of
+    the method is that it introduces non-uniformities in the camera response
+    for the real data, and therefore data-MC discrepancies (there are no stars
+    in MC and all pixels have the same cleaning settings).
+
+    Here we try to calculate what should be the "base" picture threshold (to
+    use both in MC and data) so that at most a given fraction "max_fraction"
+    of the camera gets an increased threshold via the
+    clean_with_pedestal_threshold condition. In this way the number /
+    extension of camera inhomogeneities in the real data is limited. By
+    default the max_fraction is 0.04, which e.g. gives a picture threshold
+    around 8 for the Crab field (in no-moon conditions).
+
+    Note: we want cleaning settings for a whole run, so we will have to run
+    the function over sub-runs and average the values, or make the function
+    able to read multiple DL1 files
+
+    Parameters
+    ----------
+    dl1_path: real data DL1 file, to have access to the monitoring table
+    where we can read the pedesta bias & std dev estimated with interleaved
+    pedestal events
+
+    config_file: must be the one used for the analysis of the real data,
+    in particular, it has to contain the tailcuts_clean_with_pedestal_threshold
+    setting (sigma) that one wants to use
+
+    max_fraction: maximum fraction of camera pixels that are allowed to get a
+    picture threshold above the base one. That is, we calculate here the base
+    picture threshold that will ensure that the condition is fulfilled
+
+    Returns: scalar
+    the value of the picture threshold that has to be used in data and
+    MC to ensure that no more than max_fraction of the camera gets an
+    increased value via tailcuts_clean_with_pedestal_threshold
+
+    -------
+
+    """
+    std_config = get_standard_config()
+    if config_file is not None:
+        config = replace_config(std_config,
+                                read_configuration_file(config_file))
+    else:
+        config = std_config
+
+    cleaning_method = 'tailcuts_clean_with_pedestal_threshold'
+    picture_th, _, _, _ = get_cleaning_parameters(config, cleaning_method)
+    sigma = config[cleaning_method]['sigma']
+
+    # Obtain the picture thresholds of pixels based on the "clean with
+    # pedestal threshold" method:
+    pic_threshold = get_threshold_from_dl1_file(dl1_path, sigma)
+    threshold_sorted = np.sort(pic_threshold)
+
+    # find the value new_threshold above which a fraction max_fraction of
+    # pixels lies:
+    index = int(len(threshold_sorted) * (1 - max_fraction))
+    new_threshold = threshold_sorted[index]
```

Comment:
Maybe `np.percentile` could improve this. Using `interpolation=lower/upper` should yield the desired result and avoid the need to sort the array completely

Reply:
Performance is not an issue here, it is not a large array. I prefer not to touch it, because I am very able to make it stop working even with such an easy change...